src/components/pages/maintenance/ruleDetails/ruleDetails.js

```diff
@@ -132,17 +134,24 @@ export class RuleDetails extends Component {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
 
-  // TODO: Handle error and isPending states
-  updateAlertStatus = (selectedAlerts, status) =>
+  // TODO: Handle error states
+  updateAlertStatus = (selectedAlerts, status) => {
+    this.setState({
+      isAlertStatusPending: true,
+      alertContextBtns: this.getAlertContextBtns(selectedAlerts, true)
```

Comment:
Why are you doing this? It should be as simple as setting the flag and checking it to render the indicator in the `render` method inside the `ContextMenu` component.

Reply:
You'd think that, right?  But when I did it that way... the indicator never showed up... even when I added a 5 second delay.